File: AtCoder/abc/abc302/d.cpp
#include <iostream>
#include <algorithm>
using namespace std;

typedef long long ll;

int main() {
    ll N, M, D;
    cin >> N >> M >> D;
    vector<ll> A(N), B(M);
    for (ll& a : A) cin >> a;
    for (ll& b : B) cin >> b;
    sort(A.begin(), A.end(), greater<ll>());
    sort(B.begin(), B.end(), greater<ll>());

    ll max_value = -1;
    int i = 0, j = 0;
    while (i < N && j < M) {
        cout << i << " " << j << " " << A[i] << " " << B[j] << endl;
        if (abs(A[i] - B[j]) <= D) {
            max_value = max(max_value, A[i] + B[j]);
            ++i;
            ++j;
        } else if (A[i] > B[j]) {
            ++j;
        } else {
            ++i;
        }
    }

    cout << max_value << endl;

    return 0;
}

File: AtCoder/abc/abc302/d.py
def main():
    n, m, d = map(int, input().split())

    a = list(map(int, input().split()))
    b = list(map(int, input().split()))

    a.sort(reverse=True)
    b.sort(reverse=True)
    i = 0
    j = 0
    res = -1

    while i < n and j < m:
        if abs(a[i] - b[j]) <= d:
            res = max(res, a[i] + b[j])
            i += 1
            j += 1
        elif a[i] > b[j]:
            i += 1
        else:
            j += 1
    print(res)

if __name__ == '__main__':
    main()
